### ISSUE TITLE
Libre Audio

### ISSUE THREAD

**Issue by 100best** [yvt/openspades#424]
What about replacing the audio in DevPaks with libre ones? :+1:

**Comment by yvt** [comment]
@100best Can you point me to the link on "libre ones"?

p.s. There's a branch named `fix-up` that aims to [make OS independent of proprietary assets](https://github.com/yvt/openspades/tree/fix-up/Resources/Sounds)

**Comment by Kurtoid** [comment]
Btw, where do the current sounds come from?

**Comment by feikname** [comment]
@Kurtoid They're downloaded during the build process. 

I think its from yvt's Dropbox account, but i am not sure of it.

**Comment by Kurtoid** [comment]
I know that, but where did he get them from?

**Comment by yvt** [comment]
Most of the current sounds are made from commercial sound libraries (which explains the exception terms in [the license text](http://openspades.yvt.jp)). The others are created with Logic Pro, except that `BlockFall.wav` comes from AoS 0.75's `debris.wav`.

**Comment by 100best** [comment]
What about replacing the files with Free/Libre replacements and putting then in Free/Libre formats like Opus or Vorbis?

**Comment by 100best** [comment]
@yvt Would the current sounds be free/libre if they where converted into a free/libre audio format or not?

**Comment by yvt** [comment]
@100best Unlikely.

**Comment by noway421** [comment]
@100best no.

the format they are stored in can be libre, and that would be a good idea. something like FLAC would be nice, or anything with a decent bitrate really. That would mean that it would be possible to compile openspades using only libraries which implement open source audio formats. A nice thing to have, although not essential (there are free libraries for mp3 and other formats, it's just that the formats themselves are patented)

if you want libre sounds, you can get a recorder somewhere or even a microphone or whatever works and record the needed sounds. then open up a feature branch and replace them there. A similar thing to what OpenTTD did with their proprietary pixel art assets.

**Comment by roboman2444** [comment]
How would i go about getting the "old" sounds... The ones that were used up until a few months ago?

**Comment by 100best** [comment]
Can there be added an option to not have to compile OpenSpades with `pak000-Nonfree.pak ` for OpenSpades to work cause I don't like using non-free software, also what is the license for the Models folder in `pak000-Nonfree.pak ` since I know the Sounds folder is non-free.

**Comment by yvt** [comment]
OS mostly relies on original player and map object models, which are extracted from the original client and are packaged as a part of `pak000-Nonfree.pak`. Even now it's possible to compile it without `pak000-Nonfree.pak` (by running `make` on each target but not `OpenSpades_Resources_DevPaks`), and to boot the compiled binary almost successfully, but it won't be really playable.

Placeholder files for GPL version of such game assets can be identified [with the suffix `.weak`](https://github.com/yvt/openspades/tree/master/Resources/Models/Weapons/SMG).

**Comment by 100best** [comment]
It currently works without these in `pak000-Nonfree.pak`.
```
./Sounds
./Models/Weapons/Rifle/Tracer.kv6
./Models/Weapons/Shotgun/Tracer.kv6
./Models/Weapons/SMG/WeaponNoMagazine-old.kv6
./Models/Weapons/SMG/Tracer.kv6
./Models/Weapons/Block/Block.kv6
./Models/'untitled folder'
```
It would be nice if OpenSpades could use the system installed fonts instead of depending on the ones in "pak/zip" `pak000-Nonfree.pak` and `font-unifont.pak` also if OpenSpades could use "`.tar.gz`", also OpenSpades should eventually replace the files in `./Models` and `./Sounds` also `./Gfx/fonts` and anywhere else with [GPLv3](https://www.gnu.org/licenses/gpl.html) compatible replacements, I think `./Gfx/fonts` and `./Models` are important because OpenSpades is currently playable without `./Sounds ` but not without the other two.

**Comment by yvt** [comment]
I agree about replacing models with GPLv3 compatible replacements, since the license uncertanity of these files is causing problems among package maintainers. Regarding fonts, do they really need to be GPLv3 compatible? They're already SIL OFL. Looks like [SuperTuxKart has OFL font assets](https://github.com/supertuxkart/stk-code/blob/7c2642d42769b3dff7e800e0d92b2b4a061147ea/data/ttf/LICENSE), too.

**Comment by NotAFile** [comment]
What is the current status of licenses for non-free assets in OpenSpades?
Does OS have permission to use the AoS assets?

**Comment by yvt** [comment]
I consider that it's currently in a gray zone as long as the original author is clearly credited. The reason for this is that he didn't mention its license and the original client didn't include any EULA or something that could impose conditions on the redistribution of the game assets. Furthermore, considering that the original author [hasn't been around for several years](https://www.reddit.com/r/AceOfSpades/comments/2det6o/any_trace_of_ben_aksoy/), it can be said that the original client is practically an [abandonware](https://en.wikipedia.org/wiki/Abandonware).

Anyway, the original author has a right on the game assets. If he's suddenly back after several years of silence, he will have the right on our fate.

Also, It's also possible that the copyright extends to entire the game design. That's why I can't submit this game to Greenlight even if all game assets were replaced.

**Comment by NotAFile** [comment]
How much does the game need to be different to be able to submit it to greenlight? I presume submitting it there eventually is the goal, since it's pretty much the only way we could gain traction for it, considering the sad state the community is in currently

**Comment by noway** [comment]
Can it be argued that game design is different even though it uses on the same protocol? One simple change (addition/extension, for example an extra packet understood only by OS) can be argued to drastically change the design.

is protocol copyrighted though?

**Comment by yvt** [comment]
> Can it be argued that game design is different even though it uses on the same protocol? One simple change (addition/extension, for example an extra packet understood only by OS) can be argued to drastically change the design.

Yes, it's technically possible, but is hard to achieve. Minor modifications are likely to end up as the original game design's derivative work, which are considered still owned by the original author.

> is protocol copyrighted though?

Protocol isn't copyrighted since it's not an artistic creation by its own.

**Comment by noway** [comment]
Does the protocol have any Intellectual Property rights? I haven't heard of it being patented, so most likely besides that there are no possible way to impose IP on it. Which is good and explains why pyspades had no problems whatsoever with Ken and later Jagex.

Steam seems to be having no information on possibility of listing derivative works there. The only info about IP is this:
>You own the rights to sell the game you are posting, or you have specific authorization to represent the developer


>Additionally, you agree not post any item to Greenlight that contains the following:


>Games using copyright material such as assets or intellectual property without permission from the owner

Although Jagex completely changed the game design in the new version, I bet they still own the rights on the beta game design, as because they bought out the game from Ken as a whole..

Hmm, would be nice to get in touch with Ken and ask him under what conditions the deal went through. There might be some legal quirks there. And by the way, why did Jagex decide to remake the game to begin with, was there a legal reason to it maybe or simply the NIH syndrome?

**Comment by NotAFile** [comment]
Jagex didn't start the remake I think, mat^2 (and ben?) had built an opengl client before being bought. (https://youtu.be/p3jPXgsXPVE) It was remade because it was built on an extremely old engine with a restrictive license, and had CPU only rendering. It was an obvious choice to rewrite it.

It looks like the way it currently is, OpensSpades is definitely not in a good spot legally. I think the only real way to fix this would be to get the rights.

**Comment by DeathByDenim** [comment]
Hi there,

I'm just wondering what is needed for the final push to make OpenSpades truly libre. Do you only need new asset replacements?

If so, which assets? From the discussion above, I gather it's mostly the audio files that need to be replaced or is there more to it? Possibly the fonts?

**Comment by DeathByDenim** [comment]
I made replacements for all of the assets in pak000-NonFree.pak so that OpenSpades can be fully libre.
I'm not entirely sure how where to do a pull request for this. In any case, see https://github.com/DeathByDenim/openspades-free-pak for the replacement.

The release section contains a 001-Free.pak which has libre alternatives for all the assets in 000-NonFree.pak. Almost all assets are licensed under CC0 with a few exceptions that are CC-BY 3.0 or 4.0. All sources are in the licence.md file.

**Comment by DeathByDenim** [comment]
Following up on my earlier comment, would there be interest in substituting the current non-free assets by the free ones I made to make OpenSpades truly open? The licence for the free assets allow you to just grab them.

**Comment by RustoMCSpit** [comment]
will this be merged soon?
